lib/RAST_SDK/RAST_SDKImpl.pm

```diff
@@ -1540,18 +1540,26 @@ sub annotate_genomes
 		}
 	}
 		
-		my $output_genomeset;
-		if (defined $params->{output_genome} && $params->{output_genome} gt ' ') { 
-			my $output_genomeset = $params->{output_genome};
-	        my $genome_set_name = $params->{output_genome};
-	        my $genome_set = Bio::KBase::kbaseenv::su_client()->KButil_Build_GenomeSet({
-	            workspace_name => $params->{workspace},
-	            input_refs => $output_genomes,
-	            output_name => $output_genomeset,
-	            desc => 'GenomeSet Description'
-	        });
-		}
-
+    my $output_genomeset;
+    if (defined $params->{output_genome} && $params->{output_genome} gt ' ') {
+        my $output_genomeset = $params->{output_genome};
+        my $report_reference = Bio::KBase::kbaseenv::su_client()->KButil_Build_GenomeSet({
+            workspace_name => $params->{workspace},
+            input_refs => $output_genomes,
+            output_name => $output_genomeset,
+            desc => 'GenomeSet Description'
+        });
+
+        my $newset = Bio::KBase::kbaseenv::get_objects([{
+            workspace => $params->{workspace},
+            name => $output_genomeset
+        }]);
+        my $info = $newset->[0]->{info};
+        Bio::KBase::kbaseenv::add_object_created({
+            "ref" => $info->[6]."/".$info->[0]."/".$info->[4],
```

Comment:
You can probably actually get away with `$params->{workspace}."/".$output_genomeset and avoid looking up the object entirely. These refs are converted to UPAs when saved in the WS

Reply:
Where so you get the documentation for this? I'm just copying other code that I've seen. I don't know where to find the no_data option.